internal/transport/transport_test.go

```diff
@@ -1248,6 +1254,62 @@ func (s) TestServerWithMisbehavedClient(t *testing.T) {
 	}
 }
 
+func (s) TestClientHonorsConnectContext(t *testing.T) {
+	// Create a server that will not send a preface.
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening: %v", err)
+	}
+	defer lis.Close()
+	go func() { // Launch the misbehaving server.
+		sconn, err := lis.Accept()
+		if err != nil {
+			t.Errorf("Error while accepting: %v", err)
+			return
+		}
+		defer sconn.Close()
+		if _, err := io.ReadFull(sconn, make([]byte, len(clientPreface))); err != nil {
+			t.Errorf("Error while reading client preface: %v", err)
+			return
+		}
+		sfr := http2.NewFramer(sconn, sconn)
+		// Do not write a settings frame, but read from the conn forever.
+		for {
+			if _, err := sfr.ReadFrame(); err != nil {
+				return
+			}
+		}
+	}()
+
+	// Test context cancelation.
+	timeBefore := time.Now()
+	connectCtx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Could we please use `defaultTestTimeout` which is defined for this package instead of a hard-coded value. We usually end up with different values in different tests over time. Thanks.

Reply:
Done.  If it ever changes to a value less than 3s then this test becomes broken, though, so it is more sensitive to the value than most other tests.